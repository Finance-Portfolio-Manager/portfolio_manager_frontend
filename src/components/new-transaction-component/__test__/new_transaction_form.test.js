import React from 'react';
import renderer from 'react-test-renderer';
import NewTransactionForm from '../new_transaction_form.js';

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = shallow(<NewTransactionForm></NewTransactionForm>);

test('matches snapshot', () => {
    const component = renderer.create(
      <NewTransactionForm></NewTransactionForm>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('4 buttons render', ()=>{
  expect(wrapper.find("Button").length).toBe(4);
})

test("5 labels render", ()=>{
  expect(wrapper.find("FormLabel").length).toBe(5);
})