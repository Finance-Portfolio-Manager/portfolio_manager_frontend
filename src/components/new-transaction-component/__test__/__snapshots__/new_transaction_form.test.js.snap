// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="wholeform"
>
  <form
    className=""
  >
    <div
      class="container"
    >
      <button
        className="btn btn-info"
        disabled={false}
        name="buy"
        type="button"
        value="buysell"
      >
        Buy
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        name="Sell"
        type="button"
        value="buysell"
      >
        Sell
      </button>
    </div>
    <div
      class="container"
    >
      <div>
        <label
          className="form-label"
        >
          Stock Symbol:
        </label>
        <input
          className="form-control"
          name="stocksymbol"
        />
        <label
          className="form-label"
        >
          Quantity:
        </label>
        <input
          className="form-control"
          name="shareAmount"
          type="number"
        />
        <label
          className="form-label"
        >
          Price:
        </label>
        <input
          className="form-control"
          name="sharePrice"
          type="number"
        />
        <label
          className="form-label"
        >
          Date:
        </label>
        <input
          className="form-control"
          name="date"
          type="date"
        />
        <label
          className="form-label"
        >
          Note:
        </label>
        <input
          className="form-control"
          name="note"
        />
      </div>
    </div>
    <div
      class="container"
    >
      <button
        className="btn btn-info"
        disabled={false}
        type="submit"
      >
        Add New Transaction
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        type="submit"
      >
        Test Get Transaction
      </button>
    </div>
  </form>
</div>
`;
